RDoc-1909_IndexThrottling (#1358)
ravendb/docs: Documentation/5.2/Raven.Documentation.Pages/indexes/index-throttling.markdown

tasks and the procession of larger batches when indexing does take place.  

{INFO: }
Throttled indexes are kept [stale](../indexes/stale-indexes#indexes-stale-indexes) 
as long as their procession is incomplete.  
They are therefore expected to remain stale longer than indexes that are processed sequentially.  
{INFO/}

{PANEL/}

{PANEL: Throttling and Batches}

The delay in index procession increases the size of Item [batches](../server/configuration/indexing-configuration#indexing.mapbatchsize) 
that are processed during indexing intervals.  

In special cases, throttled indexes' batches are processed without delay.  

* The batch following a batch that cannot be processed for known reasons, will be processed 
  without delay.  
* Accumulated data (collected due to batch size configuration, for example) whose procession 
  requires multiple batches, will be processed with no delay between batches.  

{PANEL/}

{PANEL: Setting Index Throttling}

---

### Server-Wide Index Throttling

 Index Throttling can be set server-wide using a designated [configuration option](../server/configuration/configuration-options#json).  

>  Setting the server-wide configuration option will apply to all databases on all nodes.  

Reviewer: ```suggestion
 Setting the server-wide configuration option will apply to all databases on a given node. 
```

Currently it sounds like setting it on a single node will make that it will be applied "on all nodes". That's not the case. If you set it on a given node then only this node will have it.

Author: done

      3. **Set Default** - Click 'Set Default' to select the server-wide default value.  
      4. **Save** - Apply changes.  
         {WARNING: }
         An edited configuration key's value will become effective only after the database is reloaded.  
         {WARNING/}

---

### Index Throttling in an Index Definition

Set throttling for a specific index using the `Indexing.ThrottlingTimeInterval` property.  
Setting this property overrides [server-wide](../indexes/index-throttling#server-wide-index-throttling) 
and [database](../indexes/index-throttling#index-throttling-per-database) settings configuration.  
  
{CODE-BLOCK:csharp}
var index = new Orders_ByOrder
{
    Configuration =
    {
        [RavenConfiguration.GetKey(x => x.Indexing.ThrottlingTimeInterval)] = "2000",
    }
};
await index.ExecuteAsync(store);
{CODE-BLOCK/}

The batch size can be left for RavenDB to decide, or you can set it yourself using 
the [MapBatchSize](../server/configuration/indexing-configuration#indexing.mapbatchsize) property.  
{CODE-BLOCK:csharp}
[RavenConfiguration.GetKey(x => x.Indexing.MapBatchSize)] = "500"
{CODE-BLOCK/}

> 

Reviewer: I think it would be good to show how to setup it via the studio as well:

![image](https://user-images.githubusercontent.com/376551/123592881-20f44600-d7ee-11eb-9327-8c18225d2928.png)

Author: done

{PANEL/}

## Related Articles

### Indexes

> - [auto Indexes](../indexes/creating-and-deploying#auto-indexes)  

Reviewer: ```suggestion
- [Auto Indexes](../indexes/creating-and-deploying#auto-indexes)  
```

Author: fixed

> - [static Indexes](../indexes/creating-and-deploying#static-indexes)  

Reviewer: ```suggestion
- [Static Indexes](../indexes/creating-and-deploying#static-indexes)  
```

Author: fixed

### Server

> - [configuration options](../server/configuration/configuration-options#json)  

Reviewer: ```suggestion
- [Configuration options](../server/configuration/configuration-options#json)  
```

Author: fixed
